client/task_runner.go

```diff
@@ -175,25 +184,62 @@ func (r *TaskRunner) Run() {
 		}
 	}
 
+	// Monitoring the Driver
+	err = r.monitorDriver(r.handle.WaitCh(), r.updateCh, r.destroyCh)
+	for err != nil {
+		r.logger.Printf("[ERR] client: failed to complete task '%s' for alloc '%s': %v",
+			r.task.Name, r.allocID, err)
+		r.restartTracker.increment()
+		shouldRestart, when := r.restartTracker.nextRestart()
+		if !shouldRestart {
+			r.logger.Printf("[INFO] Not restarting")
```

Comment:
I think you could remove this log line as the alloc status changes making it redundant